File: flipping-out/app/internal/models/feature_flags.go
package models

type FeatureFlag struct {
	ID          string             `bson:"_id,omitempty" json:"id,omitempty"`
	Flag        string             `bson:"flag" json:"flag"`
	Variations  VariationSet       `bson:"variations" json:"variations"`
	DefaultRule FeatureFlagRuleSet `bson:"defaultRule" json:"defaultRule"`
}

type VariationSet struct {
	DefaultVar bool `bson:"default_var" json:"default_var"`
	FalseVar   bool `bson:"false_var" json:"false_var"`
	TrueVar    bool `bson:"true_var" json:"true_var"`
}

type FeatureFlagRuleSet struct {
	Percentage map[string]int `bson:"percentage" json:"percentage"`
}

File: flipping-out/client/app/types/feature-flags.ts
export type FeatureFlag = {
  id: string;
  flag: string;
  variations: {
    default_var: boolean;
    false_var: boolean;
    true_var: boolean;
  };
  defaultRule: {
    percentage: {
      false_var: number;
      true_var: number;
    };
  };
};
